### PR TITLE
Add option for for enabling or disabling writing BAI and SBI files.

### PR DESCRIPTION
**Pull request by tomwhite**
Relies on https://github.com/disq-bio/disq/pull/69, which has not yet been merged or released.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java [hunk 1]
```diff
@@ -119,6 +121,16 @@
             optional = true)
     protected int numReducers = 0;
 
+    @Argument(doc = "For tools that write output BAM, write an BAI file. Defaults to true.",
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: This should use the same argument name / configuration as this:

https://github.com/broadinstitute/gatk/blob/1f6a172a34cf69841902278fa05b16669fb8cc4a/src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java#L75-L78

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java [hunk 2]
```diff
@@ -119,6 +121,16 @@
...
+            fullName = WRITE_SBI_LONG_NAME,
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: woops, copy/paste error here

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java [hunk 3]
```diff
@@ -78,6 +78,8 @@
     public static final String NUM_REDUCERS_LONG_NAME = "num-reducers";
     public static final String SHARDED_OUTPUT_LONG_NAME = "sharded-output";
     public static final String OUTPUT_SHARD_DIR_LONG_NAME = "output-shard-tmp-dir";
+    public static final String WRITE_BAI_LONG_NAME = "write-bai";
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: We have a constant for this already, see comment below.

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java [hunk 4]
```diff
@@ -78,6 +78,8 @@
...
+    public static final String WRITE_SBI_LONG_NAME = "write-sbi";
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: maybe rename to `create-output-bam-splitting-index` for consistency

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java (hunk 2 above)
Comment (pshapiro4broad) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: typo
```suggestion
            fullName = WRITE_BAI_LONG_NAME,
```

Comment (tomwhite) [reply src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: Fixed.

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java (hunk 1 above)
Comment (tomwhite) [reply src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: Thanks for pointing this out! Fixed.

File: src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java (hunk 4 above)
Comment (tomwhite) [reply src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java]: Done.

File: src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java [hunk 5]
```diff
@@ -120,41 +118,44 @@ public void testSpecifyPartsDir(String inputBam, String outputFileName, String r
...
+        // check that a bai file is created
+        if (IOUtils.isBamFileName(outputPath) && writeBai) {
+            Assert.assertTrue(Files.exists(IOUtils.getPath(outputPath + BAMIndex.BAMIndexSuffix)));
+        }
         // check that a splitting bai file is created
-        if (IOUtils.isBamFileName(outputPath)) {
-            // Reinstate once https://github.com/disq-bio/disq/issues/45 is done
-            //Assert.assertTrue(Files.exists(IOUtils.getPath(outputPath + SBIIndex.FILE_EXTENSION)));
+        if (IOUtils.isBamFileName(outputPath) && writeSbi) {
+            Assert.assertTrue(Files.exists(IOUtils.getPath(outputPath + BAMIndex.BAMIndexSuffix)));
```
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java]: Should be `SBIIndex.FILE_EXTENSION`. You should also permute the tests in the data provider above to include a case of `true, false` and `false, true` for extension arguments so we can be sure to test that they can be requested independently.

Comment (tomwhite) [reply src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java]: Good catch. I've fixed this now.

**Comment by tomwhite** [comment]
Thanks @jamesemery. The compile errors will be fixed once Disq 0.2.0 is released.

**Comment by tomwhite** [comment]
@lbergelson @jamesemery this is passing now and all comments have been addressed. Can you unblock so it can be merged please?

**Pull request merged**